pyoptsparse/pyOpt_optimization.py

```diff
@@ -229,71 +240,11 @@ def addVarGroup(
         if varType not in ["c", "i", "d"]:
             raise Error("Type must be one of 'c' for continuous, 'i' for integer or 'd' for discrete.")
 
-        # ------ Process the value argument
-        value = np.atleast_1d(value).real
-        if len(value) == 1:
-            value = value[0] * np.ones(nVars)
-        elif len(value) == nVars:
-            pass
-        else:
-            raise Error(
-                f"The length of the 'value' argument to addVarGroup is {len(value)}, "
-                + f"but the number of variables in nVars is {nVars}."
-            )
-
-        if lower is None:
-            lower = [None for i in range(nVars)]
-        elif np.isscalar(lower):
-            lower = lower * np.ones(nVars)
-        elif len(lower) == nVars:
-            lower = np.atleast_1d(lower).real
-        else:
-            raise Error(
-                "The 'lower' argument to addVarGroup is invalid. "
-                + f"It must be None, a scalar, or a list/array or length nVars={nVars}."
-            )
-
-        if upper is None:
-            upper = [None for i in range(nVars)]
-        elif np.isscalar(upper):
-            upper = upper * np.ones(nVars)
-        elif len(upper) == nVars:
-            upper = np.atleast_1d(upper).real
-        else:
-            raise Error(
-                "The 'upper' argument to addVarGroup is invalid. "
-                + f"It must be None, a scalar, or a list/array or length nVars={nVars}."
-            )
-
-        # ------ Process the scale argument
-        if scale is None:
-            scale = np.ones(nVars)
-        else:
-            scale = np.atleast_1d(scale)
-            if len(scale) == 1:
-                scale = scale[0] * np.ones(nVars)
-            elif len(scale) == nVars:
-                pass
-            else:
-                raise Error(
-                    f"The length of the 'scale' argument to addVarGroup is {len(scale)}, "
-                    + f"but the number of variables in nVars is {nVars}."
-                )
-
-        # ------ Process the offset argument
-        if offset is None:
-            offset = np.ones(nVars)
```

Comment:
I think this is actually a bug. We should have `offset` default to zero instead of one (see default value above). In any case I don't think we should allow user to pass in `None`.

Reply:
Agreed, looks like a copy-paste leftover. `None` does not make sense here, I agree with using the `allow_none=False` for this broadcast